util/logger.go

```diff
@@ -3,47 +3,67 @@ package util
 import (
 	"fmt"
 	"io"
-	"log"
 	"os"
-	"strings"
 
 	"github.com/fatih/color"
 	"github.com/hashicorp/hcl/v2"
 	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/sirupsen/logrus"
 	"golang.org/x/crypto/ssh/terminal"
 )
 
-// Create a logger with the given prefix
-func CreateLogger(prefix string) *log.Logger {
-	return CreateLoggerWithWriter(os.Stderr, prefix)
+const DEFAULT_LOG_LEVEL = logrus.WarnLevel
+
+// LogEntry is a global logentry for the application
+// The idea is simple: we start with global logger and DEFAULT_LOG_LEVEL, and update it's configuration if needed
+// Ideally global objects passed around are considered an antipattern, but here we can tolerate it till we migrate to 
+// proper cli library (see https://github.com/gruntwork-io/terragrunt/blob/master/cli/args.go#L29)
+var LogEntry *logrus.Entry
+
+func init() {
+	LogEntry = CreateLogEntry("", DEFAULT_LOG_LEVEL)
 }
 
-// CreateLoggerWithWriter Create a logger around the given output stream and prefix
-func CreateLoggerWithWriter(writer io.Writer, prefix string) *log.Logger {
+// CreateLogger creates a logger. If debug is set, we use ErrorLevel to enable verbose output, otherwise - only errors are shown
+func CreateLogger(lvl logrus.Level) *logrus.Logger {
+	logger := logrus.New()
+	logger.SetLevel(lvl)
+	return logger
+}
+
+// CreateLogEntry creates a logger entry with the given prefix field
+func CreateLogEntry(prefix string, level logrus.Level) *logrus.Entry {
+	logger := CreateLogger(level)
+	var fields logrus.Fields
 	if prefix != "" {
-		prefix = fmt.Sprintf("[%s] ", prefix)
+		fields = logrus.Fields{"prefix": prefix}
+	} else {
+		fields = logrus.Fields{}
 	}
-	return log.New(writer, fmt.Sprintf("[terragrunt] %s", prefix), log.LstdFlags)
+	return logger.WithFields(fields)
 }
 
-// MAINTAINER'S NOTE: This is a temporary solution for logging levels in terragrunt. This is not a permanent debug
-// logging solution.
-// Debugf will only print out terragrunt logs if the TG_LOG environment variable is set to DEBUG.
-func Debugf(logger *log.Logger, fmtString string, fmtArgs ...interface{}) {
-	if strings.ToLower(os.Getenv("TG_LOG")) == "debug" {
-		logger.Printf(fmtString, fmtArgs...)
+// CreateLoggerWithWriter Create a logger around the given output stream and prefix
+func CreateLogEntryWithWriter(writer io.Writer, prefix string, level logrus.Level) *logrus.Entry {
+	if prefix != "" {
+		prefix = fmt.Sprintf("[%s] ", prefix)
+	} else {
+		prefix = fmt.Sprintf("[terragrunt] %s", prefix)
 	}
+	logger := CreateLogEntry(prefix, level)
+	logger.Logger.SetOutput(writer)
+	return logger
 }
 
 // ColorLogf
-func ColorLogf(logger *log.Logger, colorCode *color.Color, fmtString string, fmtArgs ...interface{}) {
+func ColorLogf(logger *logrus.Entry, colorCode *color.Color, fmtString string, fmtArgs ...interface{}) {
```

Comment:
This can probably be removed, as mentioned in a comment earlier, and replaced with `Errorf`.